### PR TITLE
Throw error with descriptive message when a Keyless Signature is invalid due to JWK rotation

### PR DESCRIPTION
**Pull request by heliuchuan**
### Description

Sometimes a JWK can be rotated unexpectedly by the OIDC provider.  This provides more descriptive error messaging for devs/users and directs them to reauthenticate.

This also implements a standardized KeylessError object
### Test Plan

Added an e2e test
### Checklist
  - [x] Have you ran `pnpm fmt`?

### CODE REVIEW

File: src/account/AbstractKeylessAccount.ts [hunk 1]
```diff
@@ -269,6 +284,19 @@ export abstract class AbstractKeylessAccount extends Serializable implements Acc
     }
     return true;
   }
+
+  static async checkJWKRotation(args: { publicKey: KeylessPublicKey | FederatedKeylessPublicKey; kid: string }) {
```
Comment (0xmaayan) [inline src/account/AbstractKeylessAccount.ts:288]: letd add a doc string for this function

**Review by 0xmaayan — COMMENTED** [review]
LGTM. Just need to fix lint and dependency cycle errors, also pleas update CHANGELOG

File: src/core/crypto/keyless.ts [hunk 2]
```diff
@@ -764,3 +764,19 @@ async function getGroth16VerificationKeyResource(args: {
...
+ * @param jwtHeader The JWT header string
+ * @returns Parsed JWT header as an object.
+ */
+export function parseJwtHeader(jwtHeader: string): JwtHeader {
+  try {
+    return JSON.parse(jwtHeader);
+  } catch (error) {
+    throw new Error("Failed to parse JWT header.");
+  }
+}
```
[the remark this answers is not in the document]
Comment (heliuchuan) [reply src/core/crypto/keyless.ts:782]: so in this context we only have access to the header, not the token

File: src/account/MultiKeyAccount.ts [hunk 3]
```diff
@@ -28,7 +28,7 @@ export interface VerifyMultiKeySignatureArgs {
  *
  * Note: Generating a signer instance does not create the account on-chain.
  */
-export class MultiKeyAccount implements Account {
+export class MultiKeyAccount implements KeylessSigner {
```
[the remark this answers is not in the document]
Comment (heliuchuan) [reply src/account/MultiKeyAccount.ts]: discussed offline

File: src/account/AbstractKeylessAccount.ts [hunk 4]
```diff
@@ -270,6 +317,22 @@ export abstract class AbstractKeylessAccount extends Serializable implements Acc
...
+    const { publicKey, kid } = args;
+    try {
+      await fetchJWK({
+        publicKey,
+        kid,
+      });
```
[the remark this answers is not in the document]
Comment (heliuchuan) [reply src/account/AbstractKeylessAccount.ts]: Ok impld on chain fetch

**Pull request merged**